structuretoolkit/analyse/distance.py

```diff
@@ -1,7 +1,15 @@
+from typing import Optional
+from ase.atoms import Atoms
 import numpy as np
 
 
-def get_distances_array(structure, p1=None, p2=None, mic=True, vectors=False):
+def get_distances_array(
+    structure: Atoms,
+    p1: Optional[np.ndarray] = None,
```

Comment:
You can sub-hint arrays just like lists, but I tend to be cowardly and avoid it because tbh I'm never sure how the hinting differences work between `float` and `np.float`, etc.

Reply:
I like the suggestion, still I think it is beyond the scope of this pull request.